service/route_handlers.go

```diff
@@ -557,3 +560,23 @@ func HandleSamlAssertion(w http.ResponseWriter, r *http.Request, assertion *saml
 	http.Redirect(w, r, redirectURL, http.StatusFound)
 	return
 }
+
+func isWhitelisted(redirectBackBaseValue string) bool {
+	if redirectBackBaseValue == server.GetRancherAPIHost() {
+		return true
+	}
+
+	// comma separated list of whitelisted domains to redirect to
+	redirectWhitelist := server.GetRedirectWhitelist()
+	if redirectWhitelist != "" {
+		whitelistedDomains := strings.Split(redirectWhitelist, ",")
+		for _, w := range whitelistedDomains {
+			if redirectBackBaseValue == w {
+				return true
+			}
+		}
+	}
+
+	log.Errorf("Cannot redirect to anything other than whitelisted domains and rancher api host")
```

Comment:
Lets do the error handling in the callers (some may just log, some might handle it in another way) of this method and let this method only evaluate whether the passed in parameter is in the whitelist or not